.github/workflows/main.yml

```diff
@@ -0,0 +1,273 @@
+
+      - name: Install pip requirements
+        run: |
+          mkdir ./.cache && mkdir ./.cache/pip
+          chown -R $(whoami) ./.cache/pip
+          pip install -U pip setuptools wheel --cache-dir ./.cache/pip
+          pip install -r requirements/main.txt --cache-dir ./.cache/pip
+          pip install -r requirements/tests.txt --cache-dir ./.cache/pip
+          pip install -r requirements/deploy.txt --cache-dir ./.cache/pip
+          
+      - name: Setup node
+        run: |
+          curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash
+          export NVM_DIR="$HOME/.nvm" 
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+          [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
+          nvm install $NODE_VERSION
+          npm i -g npm
+          npm install -g gulp-cli
+          npm ci
+
+      - name: Run tests
+        run: |
+          python -m coverage run -m pytest --strict --postgresql-host localhost
+          python -m coverage html
+          python -m coverage report -m --fail-under 100
+        env:
+          # The hostname used to communicate with the PostgreSQL service container
+          POSTGRES_HOST: localhost
+          # The default PostgreSQL port
+          POSTGRES_PORT: 5432
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Lint
+        run: |
+          sudo apt-get update && sudo apt-get install make
+          make lint ACTIONS=true
+
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Documentation
+        run: |
+          sudo apt-get update && sudo apt-get install make
+          make docs
+  
+  deps:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Cache pip dependencies
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-pip-deps-warehouse-p
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-
+
+      - name: Set cache permissions
+        run: |
+          mkdir ~/.cache && mkdir ~/.cache/pip
+          chown -R $(whoami) ~/.cache/pip
```

Comment:
Is this necessary? The later pip-install commands in `make deps` don't use this custom cache directory.